File: dataset/ZeroShot - Copy/CLAUDE_claude-sonnet-4-5-20250929/Task103.c

#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <stdbool.h>

void checkPasswordStrength(const char* password, char* result) {
    if (password == NULL || strlen(password) == 0) {
        strcpy(result, "Weak: Password is null or empty");
        return;
    }
    
    bool hasMinLength = strlen(password) >= 8;
    bool hasUppercase = false;
    bool hasLowercase = false;
    bool hasDigit = false;
    
    for (int i = 0; i < strlen(password); i++) {
        if (isupper(password[i])) hasUppercase = true;
        if (islower(password[i])) hasLowercase = true;
        if (isdigit(password[i])) hasDigit = true;
    }
    
    if (hasMinLength && hasUppercase && hasLowercase && hasDigit) {
        strcpy(result, "Strong");
    } else {
        strcpy(result, "Weak: ");
        if (!hasMinLength) strcat(result, "Less than 8 characters. ");
        if (!hasUppercase) strcat(result, "No uppercase letter. ");
        if (!hasLowercase) strcat(result, "No lowercase letter. ");
        if (!hasDigit) strcat(result, "No digit. ");
    }
}

int main() {
    char result[200];
    
    // Test cases
    checkPasswordStrength("Pass123word", result);
    printf("Test 1: %s\\n", result);
    
    checkPasswordStrength("weak", result);
    printf("Test 2: %s\\n", result);
    
    checkPasswordStrength("NoDigits", result);
    printf("Test 3: %s\\n", result);
    
    checkPasswordStrength("alllowercase123", result);
    printf("Test 4: %s\\n", result);
    
    checkPasswordStrength("ALLUPPERCASE123", result);
    printf("Test 5: %s\\n", result);
    
    return 0;
}

File: dataset/ZeroShot/CLAUDE_claude-sonnet-4-5-20250929/Task103.java

import java.util.regex.Pattern;

public class Task103 {
    public static String checkPasswordStrength(String password) {
        if (password == null) {
            return "Weak: Password is null";
        }
        
        boolean hasMinLength = password.length() >= 8;
        boolean hasUppercase = !password.equals(password.toLowerCase());
        boolean hasLowercase = !password.equals(password.toUpperCase());
        boolean hasDigit = password.matches(".*\\\\d.*");
        
        if (hasMinLength && hasUppercase && hasLowercase && hasDigit) {
            return "Strong";
        } else {
            StringBuilder weakness = new StringBuilder("Weak: ");
            if (!hasMinLength) weakness.append("Less than 8 characters. ");
            if (!hasUppercase) weakness.append("No uppercase letter. ");
            if (!hasLowercase) weakness.append("No lowercase letter. ");
            if (!hasDigit) weakness.append("No digit. ");
            return weakness.toString().trim();
        }
    }
    
    public static void main(String[] args) {
        // Test cases
        System.out.println("Test 1: " + checkPasswordStrength("Pass123word"));
        System.out.println("Test 2: " + checkPasswordStrength("weak"));
        System.out.println("Test 3: " + checkPasswordStrength("NoDigits"));
        System.out.println("Test 4: " + checkPasswordStrength("alllowercase123"));
        System.out.println("Test 5: " + checkPasswordStrength("ALLUPPERCASE123"));
    }
}
